interfaces/cython/cantera/ctml2yaml.py

```
@@ -0,0 +1,2188 @@
+
+
+yaml.RoundTripRepresenter.add_representer(float, represent_float)
+
+
+def get_float_or_units(node: etree.Element) -> Union[str, float]:
+    """Process an XML node into a float value or a value with units.
+
+    :param node:
+        The XML node with a value in the text and optionally a units attribute.
+
+    Given an XML nodes like::
+
+        <E units="cal/mol">1000.0</E>
+        <E>1000.0</E>
+
+    this function returns, respectively::
+
+        1000.0 cal/mol
+        1000.0
+
+    where the first value is a string and the second is a float.
+    """
+    value = float(clean_node_text(node))
+    units = node.get("units", "")
+    if units:
+        units = re.sub(r"([A-Za-z])-([A-Za-z])", r"\1*\2", units)
+        units = re.sub(r"([A-Za-z])([-\d])", r"\1^\2", units)
+        return "{} {}".format(float2string(value), units)
+    else:
+        return value
+
+
+def split_species_value_string(node: etree.Element) -> Dict[str, float]:
+    """Split a string of species:value pairs into a dictionary.
+
+    The keys of the dictionary are species names and the values are the
+    number associated with each species. This is useful for things like
+    elemental composition, mole fraction mappings, coverage mappings, etc.
+
+    The algorithm is reimplemented from ``compositionMap::parseCompString`` in
+    ``base/stringUtils.cpp``.
+    """
+    text = clean_node_text(node)
+    pairs = FlowMap({})
+    start, stop, left = 0, 0, 0
+    non_whitespace = re.compile(r"\S")
+    stop_re = re.compile(r"[,;\s]")
+    while stop < len(text):
+        try:
+            colon = text.index(":", left)
+        except ValueError:
+            break
+        # \S matches the first non-whitespace character
+        # colon + 1 here excludes the colon itself from the search
+        valstart_match = non_whitespace.search(text, colon + 1)
+        if valstart_match is None:
+            # is this the right thing to do?
+            print("valstart")
```

This should presumably throw an error?

---

I don't think it should throw an error, the `break` will break out of the processing loop. I do need to delete the comment and the debugging print statement though 😄